Data.Rmd

```diff
@@ -0,0 +1,329 @@
+The following example shows memory usage before and after loading the nycflights13 package.
+You can see that memory usage doesn't change until you inspect the `flights` dataset stored inside the package.
+
+```{r}
+pryr::mem_used()
+library(nycflights13)
+pryr::mem_used()
+
+invisible(flights)
+pryr::mem_used()
+```
+
+We recommend that you include `LazyData: true` in your `DESCRIPTION` if you are shipping `.rda` files below `data/`.
+If you use `use_data()` to create such datasets, it will automatically make this modification to `DESCRIPTION` for you.
+
+In terms of advice to package users, note that lazily-loaded datasets do **not** need to be pre-loaded.
+Above, once we did `library(nycflights13)`, we could immediately access `flights`.
+There is no call `data(flights)`, because it is not necessary.
+
+There are specific downsides to `data(some_pkg_data)` calls that support a policy of only using `data()` when it is actually necessary, i.e. for datasets that would not be available otherwise:
+
+-   By default `data()`, creates objects in the user's global workspace. There is the potential to silently overwrite pre-existing objects with new values.
+-   There is also no guarantee that `data(foo)` will create exactly one object named "foo". It could create more than one object and/or objects with totally different names.
+
+### Preserve the origin story of package data {#sec-data-data-raw}
+
+Often, the data you include in `data/` is a cleaned up version of raw data you've gathered from elsewhere.
+We highly recommend taking the time to include the code used to do this in the source version of your package.
+This makes it easy for you to update or reproduce your version of the data.
+This data-creating script is also a natural place to leave comments about important properties of the data, i.e. which features are important for downstream usage in package documentation.
+
+We suggest that you keep this code in one or more `.R` files below `data-raw/`.
+You don't want it in the bundled version of your package, so this folder should be listed in `.Rbuildignore`.
+usethis has a convenience function that can be called when you first adopt the `data-raw/` practice or when you add an additional `.R` file to the folder:
+
+```{r, eval = FALSE}
+usethis::use_data_raw()
+
+usethis::use_data_raw("my_pkg_data")
+```
+
+`use_data_raw()` creates the `data-raw/` folder and lists it in `.Rbuildignore`.
+A typical script in `data-raw/` includes code to prepare a dataset and ends with a call to `use_data()`.
+
+These data packages all use the approach recommended here for `data-raw/`:
+
+-   [babynames](https://github.com/hadley/babynames)
+-   [nycflights13](https://github.com/hadley/nycflights13)
+-   [gapminder](https://github.com/jennybc/gapminder)
+
+::: callout-warning
+## Submitting to CRAN
+
+Generally, package data should be smaller than a megabyte - if it's larger you'll need to argue for an exemption.
```

Comment:
Call forward to CRAN submission chapter and general challenge of arguing your special case.